### ISSUE TITLE
[no-duplicates] incorrect report on two non-combinable ts type import statements

### ISSUE THREAD

**Issue by JounQin** [benmosher/eslint-plugin-import#2007]
```ts
import type ESLint from 'eslint'
import type { Linter, SourceCode } from 'eslint'
```

If I merge them into one statement:

```ts
import type ESLint, { Linter, SourceCode } from 'eslint'
```

TS will fail to compile

```log
packages/eslint-plugin-mdx/src/processors/options.ts:5:8 - error TS1363: A type-only import can specify a default import or named bindings, but not both.

5 import type ESLint, { Linter, SourceCode } from 'eslint'
```

**Comment by ilyub** [comment]
Same issue here:
import type Vue from "vue";
import type { VueConstructor } from "vue";

**Comment by soryy708** [comment]
```typescript
import type ESLint from 'eslint'
import type { Linter, SourceCode } from 'eslint'
```

Is invalid because you're importing ESLint (the namespace) as if it was a type.

If you change it to:

```typescript
import type * as ESLint from "eslint";
import type { Linter, SourceCode } from "eslint";
```

Then "import/no-duplicates" reports.

Reproduced on

```json
{
  "eslint": "9.18.0",
  "@typescript-eslint/parser": "8.21.0",
  "eslint-import-resolver-typescript": "3.7.0",
  "eslint-plugin-import": "2.31.0",
  "typescript": "5.7.3"
}
```

It's indeed non-combineable.
